### PR TITLE
Parameterize Thanos retention related parameters

### PR DESCRIPTION
**Pull request by spaparaju**
This PR creates Jsonnet variables for these Thanos retention related parameters: 

- retentionResolutionRaw
- retentionResolution5m
- retentionResolution1h
- disableDownsampling

### CODE REVIEW

File: services/observatorium-metrics-template.jsonnet [hunk 1]
```diff
@@ -40,6 +40,10 @@ local obs = import 'observatorium.libsonnet';
     { name: 'THANOS_COMPACTOR_CPU_LIMIT', value: '1' },
     { name: 'THANOS_COMPACTOR_CPU_REQUEST', value: '100m' },
     { name: 'THANOS_COMPACTOR_LOG_LEVEL', value: 'info' },
+    { name: 'THANOS_COMPACTOR_RETENTION_RESOULTION_RAW', value: '14d' },
+    { name: 'THANOS_COMPACTOR_RETENTION_RESOULTION_5_MINUTES', value: '1s' },
+    { name: 'THANOS_COMPACTOR_RETENTION_RESOULTION_1_HOUR', value: '1s' },
```
Comment (periklis) [inline services/observatorium-metrics-template.jsonnet:45]: nit. usually names do not include numbers, e.g. `FIVE_MINUTES`

File: services/main.jsonnet [hunk 2]
```diff
@@ -415,6 +415,10 @@ local observatorium =
         { name: 'THANOS_QUERIER_REPLICAS', value: '3' },
         { name: 'THANOS_STORE_REPLICAS', value: '5' },
         { name: 'THANOS_COMPACTOR_LOG_LEVEL', value: 'info' },
+        { name: 'THANOS_COMPACTOR_RETENTION_RESOULTION_RAW', value: '14d' },
+        { name: 'THANOS_COMPACTOR_RETENTION_RESOULTION_FIVE_MINUTES', value: '1s' },
+        { name: 'THANOS_COMPACTOR_RETENTION_RESOULTION_ONE_HOUR', value: '1s' },
+        { name: 'THANOS_COMPACTOR_DISABLE_DOWNSAMPLING', value: 'true' },
```
Comment (periklis) [inline services/main.jsonnet:421]: Is this boolean value going to work like this?

Comment (spaparaju) [reply services/main.jsonnet:421]: work like this ? This variables disables / enables down sampling of the metrics.

Comment (periklis) [reply services/main.jsonnet:421]: Does it work for you? If yes I am fine.

Comment (squat) [reply services/main.jsonnet:421]: Yes, take a look at the bottom, this is exactly the same value used before

**Review by squat — APPROVED** [review]

**Pull request merged**